posthog/models/kafka_partition_stats/sql.py

```diff
@@ -1,32 +1,43 @@
+from dataclasses import dataclass
 from posthog.clickhouse.kafka_engine import kafka_engine
-from posthog.clickhouse.table_engines import AggregatingMergeTree
+from posthog.clickhouse.table_engines import AggregatingMergeTree, ReplacingMergeTree
 from posthog.kafka_client.topics import KAFKA_EVENTS_PLUGIN_INGESTION
 from posthog.settings import CLICKHOUSE_CLUSTER, CLICKHOUSE_DATABASE
 
-CREATE_PARTITION_STATISTICS_KAFKA_TABLE = (
-    lambda monitored_topic: f"""
-CREATE TABLE IF NOT EXISTS `{CLICKHOUSE_DATABASE}`.kafka_{monitored_topic}_partition_statistics ON CLUSTER '{CLICKHOUSE_CLUSTER}'
-(
-    `uuid` String,
-    `distinct_id` String,
-    `ip` String,
-    `site_url` String,
-    `data` String,
-    `team_id` Int64,
-    `now` String,
-    `sent_at` String,
-    `token` String
-)
-ENGINE={kafka_engine(topic=monitored_topic, group="partition_statistics")}
-SETTINGS input_format_values_interpret_expressions=0, kafka_skip_broken_messages = 100;
-"""
-)
 
-DROP_PARTITION_STATISTICS_KAFKA_TABLE = (
-    lambda monitored_topic: f"""
-DROP TABLE IF EXISTS `{CLICKHOUSE_DATABASE}`.kafka_{monitored_topic}_partition_statistics ON CLUSTER '{CLICKHOUSE_CLUSTER}';
-"""
-)
+@dataclass
+class PartitionStatsKafkaTable:
+    brokers: list[str]
+    topic: str
+    consumer_group: str = "partition_statistics"
+
+    @property
+    def table_name(self) -> str:
+        return f"kafka_{self.topic}_partition_statistics"
+
+    def get_create_table_sql(self) -> str:
+        return f"""
+            CREATE TABLE IF NOT EXISTS `{CLICKHOUSE_DATABASE}`.{self.table_name} ON CLUSTER '{CLICKHOUSE_CLUSTER}'
+            (
+                `uuid` String,
+                `distinct_id` String,
+                `ip` String,
+                `site_url` String,
+                `data` String,
+                `team_id` Int64,
+                `now` String,
+                `sent_at` String,
+                `token` String
```

Comment:
I'm not sure how accurate this is (some of these values don't seem to be sent any longer, or at least not sent consistently) but I'm not sure there's any canonical definition here for what _is_ valid beyond "whatever capture is sending"? (This section is just a refactor of the existing consumer anyway. I tried to ensure the fields used in the new destination table and materialized view were only those that seem to be consistently sent.)

Reply:
I wish we could bring back
👏 Proto 👏 Buf 👏

https://github.com/PostHog/posthog/pull/10932